src/Hint/Fixities.hs

```diff
@@ -0,0 +1,79 @@
+{-
+
+Raise a warning if you have redundant brackets in nested infix expressions.
+
+<TEST>
+yes = 1 + (2 * 3) -- @Ignore 1 + 2 * 3
+yes = (2 * 3) + 1 -- @Ignore 2 * 3 + 1
+no = (1 + 2) * 3
+no = 3 * (1 + 2)
+no = 1 + 2 * 3
+no = 2 * 3 + 1
+yes = (a >>= f) >>= g -- @Ignore a >>= f >>= g
+no = (a >>= \x -> b) >>= g
+</TEST>
+-}
+
+module Hint.Fixities(fixitiesHint) where
+
+import Hint.Type(DeclHint,Idea(..),rawIdea,toSS)
+import Config.Type
+import Control.Monad
+import Data.List.Extra
+import Data.Map
+import Data.Generics.Uniplate.DataOnly
+import Refact.Types
+
+import BasicTypes (compareFixity)
+import Fixity
+import GHC.Hs
+import GHC.Util
+import Language.Haskell.GhclibParserEx.GHC.Utils.Outputable
+import OccName
+import SrcLoc
+import RdrName
+
+fixitiesHint :: [Setting] -> DeclHint
+fixitiesHint settings _ _ x =
+  concatMap (infixBracket fixities) (childrenBi x :: [LHsExpr GhcPs])
+   where
+     fixities = foldMap getFixity settings `mappend` fromList (toFixity <$> defaultFixities)
+     getFixity (Infix x) = uncurry singleton (toFixity x)
+     getFixity _ = mempty
+
+infixBracket :: Map String Fixity -> LHsExpr GhcPs -> [Idea]
+infixBracket fixities = f Nothing
+  where
+    msg = "Redundant infix bracket due to operator fixities"
```

Comment:
Final change, can we shorten this a  bit - `Redundant bracket due to operator fixities` seems sufficient.

Reply:
Done :)